### PR TITLE
YAML style issue templates

### PR DESCRIPTION
**Pull request by tommy-waltmann**
## Description

This PR updates the issue templates to the new issue forms. I have also added a form for tasks, so we now have bug reports, feature requests, and tasks.

## Motivation and Context

Github created a more appealing way of allowing users to file issues.

## How Has This Been Tested?

We cannot test this until it gets merged. I followed much of the style used in glotzerlab/hoomd-blue#1384 to be safe.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [ ] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation (if relevant).
- [ ] I have added tests that cover my changes (if relevant).
- [ ] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+name: Bug Report
+description: Report a bug to help us improve.
```
Comment (vyasr) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: We should add default labels as well. freud should be better about labeling issues in general if possible. This one should be 'bug'.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      label: freud Version
+      description: What version of freud are you using?
+      placeholder: v2.10.0
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Python Version
+      description: What version of python are you using?
+      placeholder: v3.8.2
```
Comment (vyasr) [inline .github/ISSUE_TEMPLATE/bug_report.yml:36]: Given how many people are running on supercomputing clusters with arbitrarily slow version updating, I wouldn't include a default here.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    attributes:
+      label: Error Output
+      description: Please indicate the output from the above script and what you expect the output to be.
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: freud Version
+      description: What version of freud are you using?
+      placeholder: v2.10.0
```
Comment (vyasr) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: We should update [the bumpversion config in setup.cfg](https://github.com/glotzerlab/freud/blob/master/setup.cfg#L25) to also update this file.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      label: Python Version
+      description: What version of python are you using?
+      placeholder: v3.8.2
+  - type: dropdown
+    attributes:
+      label: System Platform
+      description: What platform are you using?
+      options:
+        - MacOS
+        - Linux
```
Comment (vyasr) [inline .github/ISSUE_TEMPLATE/bug_report.yml:43]: Windows too.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  - type: dropdown
+    attributes:
+      label: Installation method
+      description: How did you install freud?
+      options:
+        - Download from PyPI
+        - Download from conda-forge
+        - built from source
```
Comment (vyasr) [inline .github/ISSUE_TEMPLATE/bug_report.yml:53]: Containers are another option that I assume is reasonable common given the containers that @joaander maintains. Is there a way to include an "other" option where users can provide more info on alternate methods of installation? That would save us from having to maintain an exhaustive list here, just a sufficiently comprehensive collection of common methods.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  - type: dropdown
+    attributes:
+      label: Developer
+      description: Who should fix this?
+      options:
+        - I can!
+        - I can, but I'll need some help and guidance.
+        - Someone else should fix this.
```
Comment (vyasr) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: Can we reword this to something that sounds friendlier and less demanding?

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 7]
```diff
@@ -0,0 +1,29 @@
+name: Feature Request
+description: Suggest a new feature for freud
+body:
+  - type: textarea
+    attributes:
+      label: Problem
```
Comment (vyasr) [inline .github/ISSUE_TEMPLATE/feature_request.yml]: Calling this "Problem" sounds odd, but given the context and that the next section is the "Description" I'm not sure what alternative to suggest. We can move forward as is, just making this comment in case someone else comes up with a better alternative.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+        label: Alternatives
+        description: Describe any alternatives you have considered.
+  - type: dropdown
+    attributes:
+      label: Developer
+      description: Who should implement the new feature?
+      options:
+        - I can!
+        - I can, but I'll need some help/guidance.
+        - Someone else will need to implement this feature.
```
Comment (vyasr) [inline .github/ISSUE_TEMPLATE/feature_request.yml]: Same as in the bug report, reword this.

**Review by vyasr — CHANGES_REQUESTED** [review]
Do you think it's worth waiting for the next HOOMD minor release to see whether there are any issues to iron out, or do you think we'll get around to another freud release first?

**Comment by tommy-waltmann** [comment]
We don't have to wait until the next hoomd minor release to see how the issue forms look. AFAIK, this is all done through github on the main/master/trunk-minor branch. @vyasr you should be able to try and file an issue in hoomd and see how the new forms look right now.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (tommy-waltmann) [reply .github/ISSUE_TEMPLATE/bug_report.yml:36]: AFAIK, the placeholder isn't a default option, its something that is shown in faded text to show users how to format their response. Try filing a hoomd issue so you can see how these look.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 5 above)
Comment (tommy-waltmann) [reply .github/ISSUE_TEMPLATE/bug_report.yml:53]: I can add a container option, but I think those 4 options form a comprehensive set. Can you think of another way a user could install/use freud that doesn't fall under those categories?

I can add these options as placeholders in a text area if we need to allow more flexibility.

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 7 above)
Comment (tommy-waltmann) [reply .github/ISSUE_TEMPLATE/feature_request.yml]: I mostly tried to copy the formatting from the previous issue templates, where the first section was something like "What is the problem this feature is trying to solve?". Feel free to suggest another way of wording the title.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 6 above)
Comment (tommy-waltmann) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: I thought this sounded friendlier than the typical "me/someone else" options, do others think this sounds demanding?

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
Comment (joaander) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
description: Report a bug in freud.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (joaander) [reply .github/ISSUE_TEMPLATE/bug_report.yml:36]: Yes, the placeholder disappears when the user clicks in the text box.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  - type: textarea
+    attributes:
+      label: Description
+      description: A clear and concise description of the bug.
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Steps to Reproduce
+      description: Please provide a *minimal* script to reproduce the bug.
```
Comment (joaander) [inline .github/ISSUE_TEMPLATE/bug_report.yml:14]: Consider `render: python` here so the user automatically enters text in a code block.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 5 above)
Comment (joaander) [reply .github/ISSUE_TEMPLATE/bug_report.yml:53]: The real question is: If a user installed freud via some other method, would we even want to support it? Could we, if the binary build is out of our control?

Yes, add the `glotzerlab-software` container to the list.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 6 above)
Comment (joaander) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: The language I used in HOOMD is:
```
        - I will submit a pull request that fixes this problem.
        - Would someone else please fix this?
```

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 7 above)
Comment (joaander) [reply .github/ISSUE_TEMPLATE/feature_request.yml]: I use "Description" in HOOMD.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 10]
```diff
@@ -0,0 +1,29 @@
+name: Feature Request
+description: Suggest a new feature for freud
+body:
+  - type: textarea
+    attributes:
+      label: Problem
+      description: Is your feature request related to a problem? Please describe.
```
Comment (joaander) [inline .github/ISSUE_TEMPLATE/feature_request.yml]: "problem" is mentioned again here. For HOOMD, I use the language "What new capability would you like in HOOMD-blue?"

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 8 above)
Comment (joaander) [reply .github/ISSUE_TEMPLATE/feature_request.yml]: In HOOMD, I use:
```
        - I will submit a pull request that implements this feature.
        - Would someone else please implement this?
```

**Review by joaander — CHANGES_REQUESTED** [review]
Also add `.github/ISSUE_TEMPLATE/config.yml`. In it:
- Choose whether to allow blank issues or not.
- Populate the `contact_links:` section.

**Comment by joaander** [comment]
> We don't have to wait until the next hoomd minor release to see how the issue forms look. AFAIK, this is all done through github on the main/master/trunk-minor branch. @vyasr you should be able to try and file an issue in hoomd and see how the new forms look right now.

Unfortunately, I changed the default branch on hoomd to `trunk-minor`, so the forms disappeared. They will come back after the next minor release. You can look at GSD as an example: https://github.com/glotzerlab/gsd/issues/new/choose

**Comment by vyasr** [comment]
> Unfortunately, I changed the default branch on hoomd to `trunk-minor`, so the forms disappeared. They will come back after the next minor release. You can look at GSD as an example: https://github.com/glotzerlab/gsd/issues/new/choose

Right, I thought that with the choice of default branch we wouldn't see it. Thanks for pointing out that GSD has it, though.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (vyasr) [reply .github/ISSUE_TEMPLATE/bug_report.yml:36]: Thanks for that, I didn't realize how the placeholder works. I see it now.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 5 above)
Comment (vyasr) [reply .github/ISSUE_TEMPLATE/bug_report.yml:53]: I'm fine with not adding anything other than containers. The containers (specifically `glotzerlab-software`) should be there.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 6 above)
Comment (vyasr) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: I think that sounds nicer. Saying that someone else *should* fix it is what sounds demanding to me. It's a command instead of a request.

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 7 above)
Comment (vyasr) [reply .github/ISSUE_TEMPLATE/feature_request.yml]: There is also a "Description" section below, though. Otherwise I would also have suggested "Description" here.

As I said, I don't have a great alternative to suggest so if neither of you can come up with one it's fine with me if to move forward as is.

Comment (tommy-waltmann) [reply .github/ISSUE_TEMPLATE/feature_request.yml]: I may just reformat the first few sections here, we don't have to strictly follow the wording/section layout that was used before.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+        - built from source
+  - type: dropdown
+    attributes:
+      label: Developer
+      description: Who should fix this?
+      options:
+        - I will submit a pull request that fixes this problem.
+        - Would someone else please fix this?
+      validations:
+        required: true
```
Comment (joaander) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
```
On second though, let's remove this section. Users that plan to implement a fix can mention so in the comments.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        description: What additional information is useful to understand this request?
+  - type: dropdown
+    attributes:
+      label: Developer
+      description: Who should implement the new feature?
+      options:
+        - I will submit a pull request that implements this feature.
+        - Would someone else please implement this?
+    validations:
+      required: true
```
Comment (joaander) [inline .github/ISSUE_TEMPLATE/feature_request.yml]:
```suggestion
```

**Review by joaander — APPROVED** [review]
Thanks! You can merge this as is or apply my suggested edits first.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 11 above)
Comment (tommy-waltmann) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: How about we just remove the required validation?

**Review by bdice — APPROVED** [review]

**Comment by bdice** [comment]
I think this is ready for merge after @vyasr approves (currently requested changes).

**Review by vyasr — APPROVED** [review]
LGTM
